formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/conversion/StmtConversionContext.kt

```diff
@@ -253,7 +254,11 @@ fun StmtConversionContext.convertMethodWithBody(
     // as we may not encounter any `return` statement in the body
     returnTarget.variable.withIsUnitInvariantIfUnit().toViperUnusedResult(linearizer)
 
-    body.checkValidity(declaration.source, errorCollector)
+    if (!body.checkValidity(declaration.source, errorCollector)) throw SnaktInternalException(
```

Comment:
This is not an internal error, since it is caused by the user. 
I don't think this part needs to be changed at all for this PR, maybe best to just leave it as is?